compiler/prelude/jsmapping.go

```diff
@@ -313,6 +313,13 @@ var $internalize = function(v, t, recv) {
     }
     return s;
   case $kindStruct:
+    if (v !== null && v !== undefined && v.constructor === Date) {
```

Comment:
There's probably a better way to do this instead of `v !== null && v !== undefined`, isn't there?

I needed it to avoid:

- TypeError: Cannot read property 'constructor' of null
- TypeError: Cannot read property 'constructor' of undefined